packages/builder/src/components/automation/SetupPanel/AutomationBlockSetup.svelte

```diff
@@ -1065,7 +1065,12 @@
                 value={inputData[key]}
               />
             {:else if value.customType === "code"}
-              <CodeEditorModal>
+              <CodeEditorModal
+                on:hide={() => {
+                  // Push any pending changes when the window closes
+                  onChange({ [key]: inputData[key] })
```

Comment:
Does this account for saving vs cancelling? Like if you close the modal by clicking the X icon or clicking outside it, would we not want to ignore the change rather than save it?

Reply:
Although I appreciate that this is how the old modal used to work, since it updated on every key stroke. So this is probably fine since that's how it's always worked.